core/src/main/scala/cell/HandlerPool.scala

```diff
@@ -205,6 +250,31 @@ class HandlerPool(parallelism: Int = 8, unhandledExceptionHandler: Throwable =>
     })
   }
 
+  /**
+   * If a cell is triggered, it's `init` method is
+   * run to both get an initial (or possibly final) value
+   * and to set up dependencies. All dependees automatically
+   * get triggered.
+   *
+   * @param cell The cell that is triggered.
+   */
+  def triggerExecution[K <: Key[V], V](cell: Cell[K, V]): Unit = {
+    if (cell.markAsRunning())
+      execute(() => {
+        val completer = cell.asInstanceOf[CellImpl[K, V]]
+        val outcome = completer.init()
+        outcome match {
+          case Outcome(v, isFinal) =>
+            try completer.put(v, isFinal) catch { case _: IllegalStateException => /* ignore, the cell has already been completed via a dependency */ }
```

Comment:
See the test "init: concurrent completion" for why this exception needs to be caught at this point. (Otherwise, execution order would influence, if user sees an exception or not.)

Reply:
I removed the test/the catch clause. For
1. We planned to stall depenedencies anyway.
2. Quasi-determinism is not violated by an exception.